includes/class-wc-gateway-ppec-client.php

```diff
@@ -1012,7 +1012,37 @@ public function get_do_reference_transaction_params( array $args ) {
 			) ),
 		);
 
-		if ( ! empty( $details['shipping_address'] ) ) {
+		if ( ! empty( $details['shipping_address'] ) &&
+			! empty( $details['shipping_address']->getName() ) &&
+			! empty( $details['shipping_address']->getStreet1() ) &&
+			! empty( $details['shipping_address']->getCity() ) &&
+			! empty( $details['shipping_address']->getCountry() ) &&
+			// If the country is the US, a zipcode is required
```

Comment:
Where is this zip code and mandatory state validation logic coming from? Just want to check we don't miss anything else that could be mandatory.

Reply:
I got all the requirements from this doc:

https://developer.paypal.com/docs/classic/api/merchant/DoReferenceTransaction-API-Operation-NVP/#ship-to-address-fields

For example, for state it says:

> It is required for transactions only if the address is in one of the following countries: Argentina, Brazil, Canada, China, Indonesia, India, Japan, Mexico, Thailand or USA. See the list of PayPal State codes.

Reply:
Ah great, straight for PayPal. Works for me :)